docs/how-to/integrate-frontend.mdx

```diff
@@ -49,6 +49,85 @@ console.log(response.text);
 
 See more in the [client generation docs](/docs/develop/client-generation).
 
+### Asynchronous state management
+
+When building something a bit more complex, you will likely need to deal with caching, refetching and data going stale.
+[TanStack Query](https://tanstack.com/query/latest) is a popular library that was built to solve exactly these problems and works
+really well together the Encore request client.
+
+Here is a simple example of using the Encore request client together with TanStack Query:
+
+```ts
+import {
+  useQuery,
+  useMutation,
+  useQueryClient,
+  QueryClient,
+  QueryClientProvider,
+} from 'react-query'
+import Client from '../encore-client'
+
+// Create a Encore client
+const encoreClient = new Client(window.location.origin);
+
+// Create a react-query client
+const queryClient = new QueryClient()
+
+function App() {
+  return (
+    // Provide the client to your App
+    <QueryClientProvider client={queryClient}>
+      <Todos />
+    </QueryClientProvider>
+  )
+}
+
+function Todos() {
+  // Access the client
+  const queryClient = useQueryClient()
+
+  // Queries
+  const query = useQuery('todos', encoreClient.todo.List)
```

Comment:
Just a heads up, the syntax in TanStack Query v5 (released just the other day) is a bit different 👀

https://tanstack.com/query/v5/docs/react/guides/migrating-to-v5#supports-a-single-signature-one-object

Reply:
Great, thanks! Will update. Latest and greatest, here we go!